### PR TITLE
Fix invalid control escapes during RegExp parsing.

### PR DESCRIPTION
**Pull request by dbatyai**
JerryScript-DCO-1.0-Signed-off-by: Dániel Bátyai dbatyai.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/parser/regexp/re-parser.cpp [hunk 1]
```diff
@@ -619,19 +623,26 @@ re_parse_next_token (re_parser_ctx_t *parser_ctx_p, /**< RegExp parser context *
...
-        ch = lit_utf8_iterator_peek_next (iter_p);
+          ch = lit_utf8_iterator_peek_next (iter_p);
 
-        if ((ch >= LIT_CHAR_ASCII_UPPERCASE_LETTERS_BEGIN && ch <= LIT_CHAR_ASCII_UPPERCASE_LETTERS_END)
-            || (ch >= LIT_CHAR_ASCII_LOWERCASE_LETTERS_BEGIN && ch <= LIT_CHAR_ASCII_LOWERCASE_LETTERS_END))
+          if ((ch >= LIT_CHAR_ASCII_UPPERCASE_LETTERS_BEGIN && ch <= LIT_CHAR_ASCII_UPPERCASE_LETTERS_END)
+              || (ch >= LIT_CHAR_ASCII_LOWERCASE_LETTERS_BEGIN && ch <= LIT_CHAR_ASCII_LOWERCASE_LETTERS_END))
+          {
+            out_token_p->value = (ch % 32);
+            lit_utf8_iterator_advance (iter_p, 1);
```
Comment (sand1k) [inline jerry-core/parser/regexp/re-parser.cpp]: Please, change this call to `lit_utf8_iterator_incr (iter_p)`, the same effect, but simpler.

Comment (dbatyai) [inline jerry-core/parser/regexp/re-parser.cpp]: Fixed.

**Comment by sand1k** [comment]
LGTM.

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**